libraries/AP_RangeFinder/AP_RangeFinder_LeddarOne.cpp

```diff
@@ -48,36 +48,64 @@ bool AP_RangeFinder_LeddarOne::detect(RangeFinder &_ranger, uint8_t instance, AP
 // read - return last value measured by sensor
 bool AP_RangeFinder_LeddarOne::get_reading(uint16_t &reading_cm)
 {
-    if (uart == nullptr) {
-        return false;
-    }
+    uint8_t number_detections;
+    LeddarOne_Status leddarone_status;
 
-    // send a request message for Modbus function 4
-    if (send_request() != LEDDARONE_OK) {
-        // TODO: handle LEDDARONE_ERR_SERIAL_PORT
+    if (uart == nullptr) {
         return false;
     }
 
-    uint32_t start_ms = AP_HAL::millis();
-    while (!uart->available()) {
-        // wait up to 200ms
-        if (AP_HAL::millis() - start_ms > 200) {
-            return false;
+    switch (modbus_status) {
+    case LEDDARONE_MODBUS_INIT:
+        // clear buffer and buffer_len
+        memset(read_buffer, 0, sizeof(read_buffer));
+        read_len = 0;
+        modbus_status = LEDDARONE_MODBUS_PRE_SEND_REQUEST;
+
+    case LEDDARONE_MODBUS_PRE_SEND_REQUEST:
+        // send a request message for Modbus function 4
+        if (send_request() != LEDDARONE_OK) {
+            // TODO: handle LEDDARONE_ERR_SERIAL_PORT
+            break;
         }
-    }
+        modbus_status = LEDDARONE_MODBUS_SENT_REQUEST;
+        last_sending_request_ms = AP_HAL::millis();
+        break;
+
+    case LEDDARONE_MODBUS_SENT_REQUEST:
+        if (uart->available()) {
+            // change mod_bus status to read available buffer
+            modbus_status = LEDDARONE_MODBUS_AVAILABLE;
+        } else {
+            if (AP_HAL::millis() - last_sending_request_ms > 200) {
```

Comment:
This 200ms and Will value that came out from where is.
It is also used in other driver.
For this 200ms, I think that the case of the driver common time-out value, it is better defined in the common header.